src/app/shell/personal-cabinet/shared-cabinet/messages/messages.component.ts

```diff
@@ -0,0 +1,224 @@
+  provider$: Observable<Provider>;
+  @Select(ChatState.chatRooms)
+  chatRooms$: Observable<SearchResponse<ChatRoom[]>>;
+
+  constructor(protected store: Store, protected matDialog: MatDialog) {
+    super(store, matDialog);
+  }
+
+  protected init(): void {
+    this.userRole = this.store.selectSnapshot<Role>(RegistrationState.role);
+
+    this.provider$
+      .pipe(filter((provider: Provider) => !!provider))
+      .subscribe((provider: Provider) => {
+        this.providerId = provider.id;
+
+        if (this.userRole == Role.provider) {
+          this.getProviderWorkshops();
+        }
+        this.getChats();
+      });
+
+    this.setListeners();
+  }
+
+  protected addNavPath(): void {
+    this.store.dispatch(
+      new PushNavPath({
+        name: NavBarName.Messages,
+        isActive: false,
+        disable: true
+      })
+    );
+  }
+
+  getProviderWorkshops() {
+    this.store.dispatch(new GetWorkshopListByProviderId(this.providerId));
+  }
+
+  getChats() {
+    this.store.dispatch(new GetUserChatRooms(this.userRole));
+  }
+
+  setListeners() {
+    this.chatRooms.entities = this.mockChatRoom;
+    this.chatRooms.totalAmount = this.mockChatRoom.length;
+
+    this.chatRooms$
+      .pipe(
+        filter((chatRooms: SearchResponse<ChatRoom[]>) => !!chatRooms),
+        takeUntil(this.destroy$)
+      )
+      .subscribe((chatRooms: SearchResponse<ChatRoom[]>) => {
+        this.chatRooms = chatRooms;
+      });
+
+    this.filterFormControl.valueChanges
+      .pipe(
+        takeUntil(this.destroy$),
+        debounceTime(200),
+        distinctUntilChanged(),
+        filter((val: string) => !!val)
+      )
+      .subscribe((val: string) => {
+        //TODO: Implement search logic
+      });
+  }
+
+  onBlock(parentId: string): void {
+    const dialogRef = this.matDialog.open(ReasonModalWindowComponent, {
+      data: { type: ModalConfirmationType.blockParent }
+    });
+    dialogRef.afterClosed().subscribe((result: string) => {
+      if (result) {
+        const providerId = this.store.selectSnapshot<Provider>(
+          RegistrationState.provider
+        ).id;
+        const blockedParent = new BlockedParent(parentId, providerId, result);
+        this.store.dispatch(
+          new BlockParent(blockedParent, EntityType[this.subRole])
+        );
+      }
+    });
+  }
+
+  onUnBlock(parentId: string): void {
+    const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: ModalConfirmationType.unBlockParent
+      }
+    });
+    dialogRef.afterClosed().subscribe((result: string) => {
+      if (result) {
+        const providerId = this.store.selectSnapshot<Provider>(
+          RegistrationState.provider
+        ).id;
+        const blockedParent = new BlockedParent(parentId, providerId);
+        this.store.dispatch(
+          new UnBlockParent(blockedParent, EntityType[this.subRole])
+        );
```

Comment:
this is teh duplication, so I think you can create a common method to show modal window and as parameter pass the appropriate typo